### PR TITLE
Allow Passing of Custom Training Arguments in Textual Splitting AI Instantiation and Fit Method

### PR DESCRIPTION
**Pull request by MohamedAmineDHIAB**
* [x] Changed: possibility to define model_name from class attributes during instantiation of `TextualFileSplittingModel`, possibility to define some training arguments when using the fit method for instance lr , weight_decay, warmup_ratio, lr_scheduler

* [x] Added: parametrized tests for `TextualFileSplittingModel` including checks for training arguments:

image

### CODE REVIEW

File: docs/sdk/tutorials/document_splitting/index_nb.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+First, define the project to be used throughout the tutorial. Replace `YOUR_PROJECT_ID` with the actual project ID.
+
+```python editable=true slideshow={"slide_type": ""} tags=["remove-cell"] vscode={"languageId": "plaintext"}
+import logging
+from konfuzio_sdk.data import Project
+
+logging.getLogger("konfuzio_sdk").setLevel(logging.ERROR)
+# we use the following project for splitting ai tests
+YOUR_PROJECT_ID = 14392
+project = Project(id_=YOUR_PROJECT_ID)
```
Comment (iftwigs) [inline docs/sdk/tutorials/document_splitting/index_nb.md:58]: you don't need to specify this line twice, in the next cell you already define it in line 64.

File: docs/sdk/tutorials/document_splitting/index_nb.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+file_splitting_model.fit()
+```
+
+**Step 3:** Propose splits for a test document.
+
+```python editable=true slideshow={"slide_type": ""} tags=["remove-output"] vscode={"languageId": "plaintext"}
+from konfuzio_sdk.trainer.file_splitting import SplittingAI
+
+splitting_ai = SplittingAI(self.file_splitting_model)
+test_document = project.test_documents[-1]
```
Comment (iftwigs) [inline docs/sdk/tutorials/document_splitting/index_nb.md:93]: if it's meaningful to use this particular document, better use the actual ID of a document in case anything changes in a project and it's no longer the last one.

File: docs/sdk/tutorials/document_splitting/index_nb.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+**Step 4:** Display the result.
+
+```python editable=true slideshow={"slide_type": ""} tags=["remove-output"] vscode={"languageId": "plaintext"}
+print(f"The Document {test_document.id_} was split into {len(proposed_splits)} sub-Documents")
+```
+
+
+### Configurable parameters
+
+Commonly configurable parameters:
```
Comment (iftwigs) [inline docs/sdk/tutorials/document_splitting/index_nb.md:171]: can we also specify which AIs these parameters refer to? since for context-aware these are not relevant

File: docs/sdk/tutorials/document_splitting/index_nb.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+Compare the version output with the requirements listed for the model on the Hugging Face Hub.
+
+
+- `learning_rate` - Step size for the optimizer. Default: `3e-5`
+- `weight_decay` - Regularization term to prevent overfitting. Default: `1e-3`
+- `lr_scheduler_type` - Learning rate decay strategy (e.g., `'linear'`, `'cosine'`). Default: `'linear'`.
+- `warmup_ratio` - Ratio of steps for warmup during training. Default: `0.0`
+- `device` - Hardware for training (e.g., `'cuda'` for GPU or `'cpu'` for CPU). Default: `'cpu'`
+
+### Conclusion
```
Comment (iftwigs) [inline docs/sdk/tutorials/document_splitting/index_nb.md:200]: please also add full blocks of code in the end so that a user can copy and paste the whole block without having to copypaste from individual steps.

File: docs/sdk/tutorials/document_splitting/index_nb.md (hunk 2 above)
Comment (MohamedAmineDHIAB) [reply docs/sdk/tutorials/document_splitting/index_nb.md:93]: I agree with you but I thought it doesn't matter which document as we dont assert the prediction if its correct or not , most importantly the test dataset is not empty which is always fulfilled in this tutorial project (that we use anyway in the splitting ai tests pipeline)

File: docs/sdk/tutorials/document_splitting/index_nb.md (hunk 4 above)
Comment (MohamedAmineDHIAB) [reply docs/sdk/tutorials/document_splitting/index_nb.md:200]: Do you mean adding one code block for the fitting and usage of each type of splitting ai ? if yes, do you know if it's possible to make a <details> section or something just to avoid overpopulating the page with sub cells and also with one big code block (I would like to put the full code block under a detail section for every splitting ai type)

Comment (MohamedAmineDHIAB) [reply docs/sdk/tutorials/document_splitting/index_nb.md:200]: Ok I thought one code block for each splitting AI type.
Alright, I will do something similar incorporating all splitting ai types in one code block.

Comment (MohamedAmineDHIAB) [reply docs/sdk/tutorials/document_splitting/index_nb.md:202]: https://github.com/konfuzio-ai/konfuzio-sdk/pull/565/commits/c003a48861c31a1d9ff60f41bd0e0ff69270aa86

File: docs/sdk/tutorials/document_splitting/index_nb.md (hunk 1 above)
Comment (MohamedAmineDHIAB) [reply docs/sdk/tutorials/document_splitting/index_nb.md]: https://github.com/konfuzio-ai/konfuzio-sdk/pull/565/commits/a63e0f2cda56209bf5762c75a478da0f8caa6f50

File: docs/sdk/tutorials/document_splitting/index_nb.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+
+In this tutorial, you learned how to split a Document using different Splitting AIs in the Konfuzio SDK. Each model provides a tailored approach depending on the data and requirements.
+
+Below is the full code to accomplish this task:
+
+```python editable=true slideshow={"slide_type": ""} tags=["remove-output"] vscode={"languageId": "plaintext"}
```
Comment (iftwigs) [inline docs/sdk/tutorials/document_splitting/index_nb.md:208]: can you also make this cell non-executable (use` tags=["skip-execution", "nbval-skip"]`)? because there's no need to run this code as it has already been run in the individual cells

Comment (MohamedAmineDHIAB) [reply docs/sdk/tutorials/document_splitting/index_nb.md:208]: Yes you are absolutely right

**Review by iftwigs — APPROVED** [review]
